guides/source/security.md

```diff
@@ -80,31 +80,35 @@ The main objective of most attackers is to make money. The underground prices fo
 
 Here are some general guidelines on sessions.
 
-* _Do not store large objects in a session_. Instead you should store them in the database and save their id in the session. This will eliminate synchronization headaches and it won't fill up your session storage space (depending on what session storage you chose, see below).
-This will also be a good idea, if you modify the structure of an object and old versions of it are still in some user's cookies. With server-side session storages you can clear out the sessions, but with client-side storages, this is hard to mitigate.
+* _Critical data should not be stored in a session_. If the user clears their cookies or closes the browser, the data will be lost.
 
-* _Critical data should not be stored in session_. If the user clears their cookies or closes the browser, they will be lost. And with a client-side session storage, the user can read the data.
+* _Sensitive data should be stored in a session_. Client can read the local storages, and copy them at will.
+
+* _Large objects should not be stored in a session_. Instead store your objects in the database and at most, save their id in the session. This will spare you from unnecessary synchronization headaches and also, requires less session storage space.
 
 ### Session Storage
 
-NOTE: _Rails provides several storage mechanisms for the session hashes. The most important is `ActionDispatch::Session::CookieStore`._
+NOTE: _Rails provides several storage mechanisms for the session hashes. The most important one is `ActionDispatch::Session::CookieStore`._
+
+Rails `ActionDispatch::Session::CookieStore` saves the session hash in a cookie on the client-side. The server retrieves the session hash from the cookie, which effectively acts as the session ID. When you use cookies as the session storage, it is good to understand how they work.
```

Comment:
It really doesn't act *as* a session ID -- it just makes one unnecessary.

Lose the "Rails" at the start of this line; it would need to be possessive, and that seems needlessly awkward when the class name can stand alone.